tests/test_addons_toupdate.py

```diff
@@ -0,0 +1,610 @@
+    runner = CliRunner()
+    _checkout_to(initdir, 'fd2f1b3f463a541c01bf611a939f6129166d72fd')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            'cfdb4d310cd04ce0ea8098586007f7d561a1c0aa',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = 'all\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_03(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - requirements file exists
+    Test case:
+        - add a new dependency in requirements file on external library
+        - compare requirements: YES
+    Expected result:
+        - /
+    """
+    runner = CliRunner()
+    _checkout_to(initdir, '95c9bcf5823ef8b5af6c8e23073331b3bbb3bd6b')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            'fd2f1b3f463a541c01bf611a939f6129166d72fd',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = '\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_04(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - requirements file exists
+    Test case:
+        - add a new non-editable dependency in requirements file on
+          odoo addon
+        - compare requirements: YES
+    Expected result:
+        - /
+    """
+    runner = CliRunner()
+    _checkout_to(initdir, '5ce68ab261be5c280318f4c1c21545c6196c8438')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            '95c9bcf5823ef8b5af6c8e23073331b3bbb3bd6b',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = '\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_05(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - requirements file exists
+    Test case:
+        - add a new editable dependency in requirements file on odoo addon
+        - compare requirements: YES
+    Expected result:
+        - /
+    """
+    runner = CliRunner()
+    _checkout_to(initdir, '8959e3f5d1c6fea33c1cfe5f395b371c7c865832')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            '5ce68ab261be5c280318f4c1c21545c6196c8438',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = '\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_06(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - requirements file exists
+    Test case:
+        - change SHA of editable dependency on odoo addon
```

Comment:
For completnesse there should be a test a test that returns more than one addon.